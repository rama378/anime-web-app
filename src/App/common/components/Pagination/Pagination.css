.Pagination {
  text-align: right;
  padding: 16px;
}
