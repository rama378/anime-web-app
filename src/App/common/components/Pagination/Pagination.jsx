import PropTypes from 'prop-types';
import { CARD_LIMIT } from '../../constants/constant';
import Button from '../../components/Button/Button';
import './Pagination.css';

const Pagination = (props) => {
  return (
    <div className='Pagination'>
      {[...Array(props.maxPage)].map((n, index) => (
        <Button className='Pagination__bullet' style={index === props.page ? 'default' : 'clear'} key={index.toString()}>{index + 1}</Button>
      ))}
    </div>
  );
}

Pagination.propTypes = {
	limit: PropTypes.number,
  page: PropTypes.number,
  maxPage: PropTypes.number,
  onChange: PropTypes.func,
};

Pagination.defaultProps = {
  limit: CARD_LIMIT,
  page: 0,
  maxPage: 1,
  onChange: () => {},
};

export default Pagination;
