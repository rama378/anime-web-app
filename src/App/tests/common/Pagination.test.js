import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Pagination from '../../common/components/Pagination/Pagination';

const renderer = new ShallowRenderer();

describe('<Pagination />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<Pagination maxPage={10} page={0} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
