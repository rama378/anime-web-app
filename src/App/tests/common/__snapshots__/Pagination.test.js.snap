// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render and match the snapshot 1`] = `
<div
  className="Pagination"
>
  <Button
    className="Pagination__bullet"
    style="default"
  >
    1
  </Button>
  <Button
    className="Pagination__bullet"
    style="clear"
  >
    2
  </Button>
  <Button
    className="Pagination__bullet"
    style="clear"
  >
    3
  </Button>
  <Button
    className="Pagination__bullet"
    style="clear"
  >
    4
  </Button>
  <Button
    className="Pagination__bullet"
    style="clear"
  >
    5
  </Button>
  <Button
    className="Pagination__bullet"
    style="clear"
  >
    6
  </Button>
  <Button
    className="Pagination__bullet"
    style="clear"
  >
    7
  </Button>
  <Button
    className="Pagination__bullet"
    style="clear"
  >
    8
  </Button>
  <Button
    className="Pagination__bullet"
    style="clear"
  >
    9
  </Button>
  <Button
    className="Pagination__bullet"
    style="clear"
  >
    10
  </Button>
</div>
`;
